csfieldguide/chapters/content/en/coding-compression/sections/general-purpose.md

```diff
@@ -43,27 +31,340 @@ Here are some of the bits from the example earlier in this chapter; you can past
+
+In step 5, the sequence is `A`, and the next character is `M`.
+But the sequence `AM` is already in the dictionary, so we replace the whole sequence with the corresponding dictionary code (`5`), and add `AMS` to the dictionary instead (as `S` is the next character).
+
+| Step | Current Sequence | Next Character | Code | New Dictionary Entry Code | New Dictionary Sequence |
+|------|------------------|----------------|------|---------------------|-------------------|
+| 6    | SA               | M              | 7    | 9                   | SAM               |
+| 7    | M                | I              | 2    | 10                  | MI                |
+| 8    | IA               | M              | 4    | 11                  | IAM               |
+| 9    | M                | -              | 2    | -                   | -                 |
+
+When we have finished encoding the string, we now have a dictionary that looks like this:
+
+| Code | Code (Binary Representation) | Sequence |
+|------|------------------------------|----------|
+| 0    | 000000000000                 | I        |
+| 1    | 000000000001                 | A        |
+| 2    | 000000000010                 | M        |
+| 3    | 000000000011                 | S        |
+| 4    | 000000000100                 | IA       |
+| 5    | 000000000101                 | AM       |
+| 6    | 000000000110                 | MS       |
+| 7    | 000000000111                 | SA       |
+| 8    | 000000001000                 | AMS      |
+| 9    | 000000001001                 | SAM      |
+| 10   | 000000001010                 | MI       |
+| 11   | 000000001011                 | IAM      |
+
+And the string has been replaced with a series of codes, each corresponding to a value in the dictionary.
+`0 1 2 3 5 7 2 4 2`
+
+To see just how much compression we achieved, let’s take a look at the binary behind the characters.
+We'll assume that the characters are coded in 8 bits each (this is typical), based on UTF-8 coding (which is the same as ASCII for English characters).
+In binary, originally our message looked like this:
+
+```
+IAMSAMSAMIAM
+
+01001001 01000001 01001101 01010011 01000001 01001101 01010011 01000001 01001101 01001001 01000001 01001101
+```
+
+The uncompressed version has 12 characters, each using 8 bits, so there are 92 bits used to store it.
+
+And this is the compressed message:
+```
+000000000000 000000000001 000000000010 000000000011 000000000101 000000000111 000000000010 000000000100 000000000010
+```
+
+The compressed version shown here uses only 9 LZW codes, but each is 12 bits each, so this version would use 108 bits. This is more than the original, although we'll see later on (check out the "Representing the codes in fewer than 12 bits" panel at the end of this section) that there's an easy trick that can be used to reduce this to just 3 or 4 bits per code, so it would use no more than 4 x 9 = 36 bits, which is less than 40% of the original size.
```

Comment:
New sentence on a new line. I still think the second sentence is too long and goes into too much detail about calculations and comparisons &ndash; it's all good information, but it'd be better to just mention the easy trick that's effective now and explain just how effective when the trick itself is explained

Reply:
Tim purposefully wanted to include this here, it's to show the significance of this algorithm as a compression method. Leaving it in.